WBooks/Views/BookIndex/BookIndexViewModel.swift

```diff
@@ -15,19 +15,35 @@ internal final class BookIndexViewModel {
     let navigationTitle = "LIBRARY"
     let tabBarTitle = "Library"
     let bookViewModels: Property<[BookViewModel]>
+    var nextPage = 1
     
-    fileprivate let _bookViewModels = MutableProperty<[BookViewModel]>([])
-    fileprivate let _bookRepository: BookRepositoryType
+
+    private let _bookRepository: BookRepositoryType
+    private var _bookViewModels = MutableProperty<[BookViewModel]>([])
     
     init(bookRepository: BookRepositoryType) {
         _bookRepository = bookRepository
-        bookViewModels = Property(_bookViewModels)
+        bookViewModels = Property<[BookViewModel]>(_bookViewModels)
+        
     }
     
-    func initializeBookRepository() {
-        _bookViewModels <~ _bookRepository.fetchBooks()
+    func getMoreBooks() {
+        _bookRepository.fetchBooks(page: nextPage)
             .map { $0.map { BookViewModel(book: $0) } }
-            .liftError()
+            .take(first: 1)
+            .startWithResult { result in
+                switch result {
+                    
+                case .success(let books):
+                    self._bookViewModels.value += books
+                    
+                case .failure(let error):
+                    print(error)
+                }
+            }
+        
+        nextPage += 1
```

Comment:
This should only be incremented if the response is successfull otherwise in case of error you loose a page